### ISSUE TITLE
Can't connect to RoR server

### ISSUE THREAD

**Issue by Goetterescu** [tritonas00/RoRServerBot#6]
I followed the guide and I got a error which I don't understand. Did I miss something in the Guide? Here is the cmd output:

```
`C:\Users\Administrator\Desktop\beholder 1>python -B services_start.py
DEBUG   ; MainThread           ; 2021-04-21 16:14:40,254; asyncio          623; Using proactor: IocpProactor
WARNING ; MainThread           ; 2021-04-21 16:14:40,270; discord.client   257; PyNaCl is not installed, voice will NOT be supported
INFO    ; MainThread           ; 2021-04-21 16:14:40,270; main             360; LOG STARTED
INFO    ; MainThread           ; 2021-04-21 16:14:40,270; config            25; Reading configuration file configuration.xml
CRITICAL; MainThread           ; 2021-04-21 16:14:40,270; config           176; You should have at least 1 global admin!
INFO    ; MainThread           ; 2021-04-21 16:14:40,270; config           243; Successfully parsed the following servers:
INFO    ; MainThread           ; 2021-04-21 16:14:40,270; config           245;    - 45.156.85.84:50259 - user: Beholder - channel 742107024338452600
DEBUG   ; MainThread           ; 2021-04-21 16:14:40,270; config            33; Configuration read.
DEBUG   ; MainThread           ; 2021-04-21 16:14:40,270; discord.client  1018; on_message has successfully been registered as an event
INFO    ; MainThread           ; 2021-04-21 16:14:40,270; discord.client   510; logging in using static token
DEBUG   ; MainThread           ; 2021-04-21 16:14:40,536; discord.http     193; GET https://discord.com/api/v7/users/@me with None has returned 401
INFO    ; MainThread           ; 2021-04-21 16:14:40,536; main             421; Starting global shutdown sequence
ERROR   ; MainThread           ; 2021-04-21 16:14:40,536; main             432;    x Found no RoRclients running...
INFO    ; MainThread           ; 2021-04-21 16:14:40,536; main             437; Global shutdown sequence successfully finished.
INFO    ; MainThread           ; 2021-04-21 16:14:40,536; discord.client   718; Cleaning up tasks.
INFO    ; MainThread           ; 2021-04-21 16:14:40,536; discord.client    94; Closing the event loop.
Exception ignored in: <function _ProactorBasePipeTransport.__del__ at 0x000001BDB50B4040>
Traceback (most recent call last):
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python39\lib\asyncio\proactor_events.py", line 116, in __del__
    self.close()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python39\lib\asyncio\proactor_events.py", line 108, in close
    self._loop.call_soon(self._call_connection_lost, None)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 746, in call_soon
    self._check_closed()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 510, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed

C:\Users\Administrator\Desktop\beholder 1>pause
Drücken Sie eine beliebige Taste . . .
````

**Comment by tritonas00** [comment]
```
GET https://discord.com/api/v7/users/@me with None has returned 401
```

401 means unauthorized error

you have something wrong with discord configuration (discord server, discord bot token, or channel ID)

**Comment by Goetterescu** [comment]
I noticed that afterwards. I did not know that I have to connect to my discord bot. Somehow another issue occurred. I managed to bring my bot online, but he can't respond to any commands.

**Comment by tritonas00** [comment]
By commands you mean from discord channel? or in game?

If the first, check the channel ID is correct

**Comment by Goetterescu** [comment]
Sorry, I'm talking about ingame commands as -countdown and -ems

**Comment by tritonas00** [comment]
If the bot joined the rorserver, those should work fine. Can you upload the whole rorbot output?

**Comment by Goetterescu** [comment]
```

C:\Users\Administrator\Desktop\beholder 1>python -B services_start.py
DEBUG   ; MainThread           ; 2021-04-21 17:11:53,444; asyncio          623; Using proactor: IocpProactor
WARNING ; MainThread           ; 2021-04-21 17:11:53,460; discord.client   257; PyNaCl is not installed, voice will NOT be supported
INFO    ; MainThread           ; 2021-04-21 17:11:53,460; main             360; LOG STARTED
INFO    ; MainThread           ; 2021-04-21 17:11:53,460; config            25; Reading configuration file configuration.xml
CRITICAL; MainThread           ; 2021-04-21 17:11:53,460; config           176; You should have at least 1 global admin!
INFO    ; MainThread           ; 2021-04-21 17:11:53,460; config           243; Successfully parsed the following servers:
INFO    ; MainThread           ; 2021-04-21 17:11:53,476; config           245;    - 45.156.85.84:50259 - user: Beholder - channel 742107024338452600
DEBUG   ; MainThread           ; 2021-04-21 17:11:53,476; config            33; Configuration read.
DEBUG   ; MainThread           ; 2021-04-21 17:11:53,476; discord.client  1018; on_message has successfully been registered as an event
INFO    ; MainThread           ; 2021-04-21 17:11:53,476; discord.client   510; logging in using static token
DEBUG   ; MainThread           ; 2021-04-21 17:11:53,710; discord.http     193; GET https://discord.com/api/v7/users/@me with None has returned 200
DEBUG   ; MainThread           ; 2021-04-21 17:11:53,710; discord.http     209; GET https://discord.com/api/v7/users/@me has received {'id': '834155246267662367', 'username': 'Beholder', 'avatar': None, 'discriminator': '6582', 'public_flags': 0, 'flags': 0, 'bot': True, 'locale': 'en-US', 'mfa_enabled': False, 'email': None, 'verified': True}
DEBUG   ; MainThread           ; 2021-04-21 17:11:53,741; discord.http     193; GET https://discord.com/api/v7/gateway with None has returned 200
DEBUG   ; MainThread           ; 2021-04-21 17:11:53,741; discord.http     209; GET https://discord.com/api/v7/gateway has received {'url': 'wss://gateway.discord.gg'}
DEBUG   ; MainThread           ; 2021-04-21 17:11:53,944; discord.gateway  326; Created websocket connected to wss://gateway.discord.gg?encoding=json&v=6&compress=zlib-stream
DEBUG   ; MainThread           ; 2021-04-21 17:11:53,944; discord.client   358; Dispatching event socket_raw_receive
DEBUG   ; MainThread           ; 2021-04-21 17:11:53,944; discord.gateway  432; For Shard ID None: WebSocket Event: {'t': None, 's': None, 'op': 10, 'd': {'heartbeat_interval': 41250, '_trace': ['["gateway-prd-main-ctt2",{"micros":0.0}]']}}
DEBUG   ; MainThread           ; 2021-04-21 17:11:53,944; discord.client   358; Dispatching event socket_response
DEBUG   ; MainThread           ; 2021-04-21 17:11:53,944; discord.client   358; Dispatching event socket_raw_send
DEBUG   ; MainThread           ; 2021-04-21 17:11:53,944; discord.client   358; Dispatching event socket_raw_send
INFO    ; MainThread           ; 2021-04-21 17:11:53,944; discord.gateway  403; Shard ID None has sent the IDENTIFY payload.
DEBUG   ; MainThread           ; 2021-04-21 17:11:54,069; discord.client   358; Dispatching event socket_raw_receive
DEBUG   ; MainThread           ; 2021-04-21 17:11:54,069; discord.gateway  432; For Shard ID None: WebSocket Event: {'t': None, 's': None, 'op': 11, 'd': None}
DEBUG   ; MainThread           ; 2021-04-21 17:11:54,069; discord.client   358; Dispatching event socket_response
DEBUG   ; MainThread           ; 2021-04-21 17:11:54,272; discord.client   358; Dispatching event socket_raw_receive
DEBUG   ; MainThread           ; 2021-04-21 17:11:54,272; discord.gateway  432; For Shard ID None: WebSocket Event: {'t': 'READY', 's': 1, 'op': 0, 'd': {'v': 6, 'user_settings': {}, 'user': {'verified': True, 'username': 'Beholder', 'mfa_enabled': False, 'id': '834155246267662367', 'flags': 0, 'email': None, 'discriminator': '6582', 'bot': True, 'avatar': None}, 'session_id': 'e67889cec76fe06259ae6c61d61a6c6d', 'relationships': [], 'private_channels': [], 'presences': [], 'guilds': [{'unavailable': True, 'id': '742107024338452600'}], 'guild_join_requests': [], 'geo_ordered_rtc_regions': ['europe', 'russia', 'us-east', 'india', 'us-central'], 'application': {'id': '834155246267662367', 'flags': 0}, '_trace': ['["gateway-prd-main-ctt2",{"micros":189661,"calls":["discord-sessions-prd-2-97",{"micros":185638,"calls":["start_session",{"micros":149805,"calls":["api-prd-main-3jlw",{"micros":143056,"calls":["get_user",{"micros":11057},"add_authorized_ip",{"micros":12012},"get_guilds",{"micros":28829},"coros_wait",{"micros":2}]}]},"guilds_connect",{"micros":1,"calls":[]},"presence_connect",{"micros":17923,"calls":[]}]}]}]']}}
DEBUG   ; MainThread           ; 2021-04-21 17:11:54,272; discord.client   358; Dispatching event socket_response
INFO    ; MainThread           ; 2021-04-21 17:11:54,272; discord.gateway  494; Shard ID None has connected to Gateway: ["gateway-prd-main-ctt2",{"micros":189661,"calls":["discord-sessions-prd-2-97",{"micros":185638,"calls":["start_session",{"micros":149805,"calls":["api-prd-main-3jlw",{"micros":143056,"calls":["get_user",{"micros":11057},"add_authorized_ip",{"micros":12012},"get_guilds",{"micros":28829},"coros_wait",{"micros":2}]}]},"guilds_connect",{"micros":1,"calls":[]},"presence_connect",{"micros":17923,"calls":[]}]}]}] (Session ID: e67889cec76fe06259ae6c61d61a6c6d).
DEBUG   ; MainThread           ; 2021-04-21 17:11:54,304; discord.client   358; Dispatching event connect
DEBUG   ; MainThread           ; 2021-04-21 17:11:54,304; discord.client   358; Dispatching event socket_raw_receive
DEBUG   ; MainThread           ; 2021-04-21 17:11:54,304; discord.gateway  432; For Shard ID None: WebSocket Event: {'t': 'GUILD_CREATE', 's': 2, 'op': 0, 'd': {'banner': None, 'verification_level': 0, 'premium_tier': 0, 'default_message_notifications': 0, 'region': 'europe', 'unavailable': False, 'nsfw': False, 'application_id': None, 'lazy': True, 'owner_id': '529768569899057153', 'premium_subscription_count': 0, 'max_video_channel_users': 25, 'roles': [{'position': 0, 'permissions_new': '6546771521', 'permissions': 104320577, 'name': '@everyone', 'mentionable': False, 'managed': False, 'id': '742107024338452600', 'hoist': False, 'color': 0}, {'position': 4, 'permissions_new': '7771520849', 'permissions': 1329069905, 'name': 'Staff', 'mentionable': False, 'managed': False, 'id': '803626851124772896', 'hoist': True, 'color': 15158332}, {'position': 3, 'permissions_new': '7620512833', 'permissions': 1178061889, 'name': 'Truckers', 'mentionable': True, 'managed': False, 'id': '803626912073121836', 'hoist': True, 'color': 9936031}, {'position': 2, 'permissions_new': '8589410303', 'permissions': 2146959359, 'name': 'Admin', 'mentionable': False, 'managed': False, 'id': '804061236979171428', 'hoist': True, 'color': 2123412}, {'position': 1, 'permissions_new': '7620517697', 'permissions': 1178066753, 'name': 'Noodle of the week', 'mentionable': True, 'managed': False, 'id': '833659189834416128', 'hoist': True, 'color': 12745742}], 'icon': '4d714c8b3d646e153efc20d3f547304c', 'name': 'Roleplay Server', 'joined_at': '2021-04-21T14:28:45.896995+00:00', 'guild_hashes': {'version': 1, 'roles': {'omitted': False, 'hash': 'LrPokmjq0Ak'}, 'metadata': {'omitted': False, 'hash': 'xIU4gwPJV4c'}, 'channels': {'omitted': False, 'hash': 'EROOJUesfT8'}}, 'mfa_level': 0, 'members': [{'user': {'username': 'Beholder', 'id': '834155246267662367', 'discriminator': '6582', 'bot': True, 'avatar': None}, 'roles': [], 'premium_since': None, 'pending': False, 'nick': None, 'mute': False, 'joined_at': '2021-04-21T14:28:45.896995+00:00', 'is_pending': False, 'hoisted_role': None, 'deaf': False}], 'vanity_url_code': None, 'channels': [{'type': 4, 'position': 2, 'permission_overwrites': [], 'name': 'Text Channels', 'id': '742107024879386725'}, {'type': 4, 'position': 3, 'permission_overwrites': [], 'name': 'Voice Channels', 'id': '742107024879386726'}, {'user_limit': 7, 'type': 2, 'rtc_region': None, 'position': 0, 'permission_overwrites': [], 'parent_id': '742107024879386726', 'name': 'General', 'id': '742107024879386728', 'bitrate': 37000}, {'type': 0, 'topic': None, 'rate_limit_per_user': 0, 'position': 6, 'permission_overwrites': [], 'parent_id': '742107024879386725', 'name': 'general', 'last_message_id': '834441500888531006', 'id': '803626755524657192'}, {'type': 0, 'topic': 'Staff only discussions', 'rate_limit_per_user': 10, 'position': 7, 'permission_overwrites': [{'type': 'role', 'id': '742107024338452600', 'deny_new': '1024', 'deny': 1024, 'allow_new': '0', 'allow': 0}, {'type': 'role', 'id': '803626851124772896', 'deny_new': '0', 'deny': 0, 'allow_new': '3072', 'allow': 3072}], 'parent_id': '742107024879386725', 'name': 'general-staff', 'last_pin_timestamp': '2021-03-20T18:29:59+00:00', 'last_message_id': '834061030970753093', 'id': '803628220867412019'}, {'type': 0, 'topic': 'Here you can inform yourself about everything that is up to date. Server related of course', 'rate_limit_per_user': 15, 'position': 0, 'permission_overwrites': [{'type': 'member', 'id': '529768569899057153', 'deny_new': '0', 'deny': 0, 'allow_new': '3072', 'allow': 3072}, {'type': 'role', 'id': '742107024338452600', 'deny_new': '805763089', 'deny': 805763089, 'allow_new': '66624', 'allow': 66624}], 'parent_id': '833658021523619850', 'name': 'news', 'last_message_id': '833678813836869672', 'id': '803628466880381009'}, {'type': 0, 'topic': 'This channel is free for all. Please use it to ask questions and answer others', 'rate_limit_per_user': 10, 'position': 8, 'permission_overwrites': [], 'parent_id': '742107024879386725', 'name': 'support', 'last_message_id': None, 'id': '803629133971980298'}, {'type': 0, 'topic': 'Here we can discuss Events and plan them', 'rate_limit_per_user': 5, 'position': 2, 'permission_overwrites': [], 'parent_id': '806565971409895464', 'name': 'event-chat', 'last_message_id': '834068123798732839', 'id': '805524537067831297'}, {'type': 4, 'position': 1, 'permission_overwrites': [], 'name': 'Server Related', 'id': '806565971409895464'}, {'type': 0, 'topic': 'Wanna share your experiences on the game server? Post your screenshots here', 'rate_limit_per_user': 0, 'position': 3, 'permission_overwrites': [], 'parent_id': '806565971409895464', 'name': 'screenshots', 'last_message_id': '834434574674559056', 'id': '807344530600099850'}, {'type': 0, 'topic': None, 'rate_limit_per_user': 0, 'position': 1, 'permission_overwrites': [], 'parent_id': '806565971409895464', 'name': 'rules', 'last_message_id': '808787012093345812', 'id': '807353132144263189'}, {'type': 0, 'topic': None, 'rate_limit_per_user': 0, 'position': 9, 'permission_overwrites': [], 'parent_id': '742107024879386725', 'name': 'meet-the-staff', 'last_message_id': '821911943823753228', 'id': '807978927221112852'}, {'type': 4, 'position': 4, 'permission_overwrites': [], 'name': 'Special', 'id': '811280522897522698'}, {'type': 0, 'topic': "This channel can be used my Staff members to interview any players. Please inform other Staff members before you're using this channel, we do not want to fight over a interview room. I will not add a schedule. This is not necessary.", 'rate_limit_per_user': 5, 'position': 11, 'permission_overwrites': [{'type': 'role', 'id': '742107024338452600', 'deny_new': '3072', 'deny': 3072, 'allow_new': '0', 'allow': 0}, {'type': 'role', 'id': '803626851124772896', 'deny_new': '0', 'deny': 0, 'allow_new': '1024', 'allow': 1024}], 'parent_id': '811280522897522698', 'name': 'interview-room', 'last_message_id': '823962559270617109', 'id': '811280630268166185'}, {'type': 4, 'position': 5, 'permission_overwrites': [], 'name': 'Non server related (NSFW?)', 'id': '815295402034069555'}, {'type': 0, 'topic': 'For non-server related media', 'rate_limit_per_user': 0, 'position': 13, 'permission_overwrites': [], 'parent_id': '815295402034069555', 'name': 'media', 'last_message_id': '832265387432738836', 'id': '815295457965244456'}, {'type': 0, 'topic': 'For non-server related discussion', 'rate_limit_per_user': 0, 'position': 14, 'permission_overwrites': [], 'parent_id': '815295402034069555', 'name': 'general', 'last_message_id': '834411798031630376', 'id': '815295544268292146'}, {'type': 0, 'topic': 'Suggest future features (not events). This is the place to report problems, too', 'rate_limit_per_user': 10, 'position': 10, 'permission_overwrites': [], 'parent_id': '742107024879386725', 'name': 'suggestions', 'last_message_id': '833414406909984768', 'id': '818063605630369792'}, {'type': 0, 'topic': None, 'rate_limit_per_user': 0, 'position': 12, 'permission_overwrites': [], 'parent_id': '811280522897522698', 'name': 'group-projects', 'last_message_id': '827661352998666301', 'id': '823284389164089364'}, {'type': 0, 'topic': 'This channel is for reporting traffic infractions. Visit: https://gigi-server.netlify.app/guidelines#po', 'rate_limit_per_user': 10, 'position': 4, 'permission_overwrites': [], 'parent_id': '806565971409895464', 'name': 'points-system', 'last_pin_timestamp': '2021-03-26T20:43:22+00:00', 'last_message_id': '825107686655000617', 'id': '823487683128066089'}, {'type': 0, 'topic': None, 'rate_limit_per_user': 0, 'position': 5, 'permission_overwrites': [], 'parent_id': '806565971409895464', 'name': 'random-chat-logs', 'last_pin_timestamp': '2021-03-23T16:51:22+00:00', 'last_message_id': '829753047445798913', 'id': '823960859654094848'}, {'type': 4, 'position': 0, 'permission_overwrites': [], 'name': 'Announcements', 'id': '833658021523619850'}, {'type': 0, 'topic': 'Noodle removals or rewarded good noodles will be announced in here. The noodle of the week will also be tagged in this room.', 'rate_limit_per_user': 0, 'position': 15, 'permission_overwrites': [{'type': 'role', 'id': '742107024338452600', 'deny_new': '2064', 'deny': 2064, 'allow_new': '66560', 'allow': 66560}], 'parent_id': '833658021523619850', 'name': 'noodle-report', 'last_message_id': None, 'id': '833658218215243776'}, {'type': 0, 'topic': 'Violations against the BTL and VCG can be reported here. They will be announced in here by Moderators, too.', 'rate_limit_per_user': 5, 'position': 16, 'permission_overwrites': [], 'parent_id': '833658021523619850', 'name': 'traffic-violations', 'last_message_id': None, 'id': '833658270921129984'}], 'discovery_splash': None, 'stage_instances': [], 'system_channel_flags': 0, 'system_channel_id': None, 'preferred_locale': 'en-US', 'voice_states': [], 'afk_channel_id': None, 'member_count': 19, 'presences': [], 'description': None, 'explicit_content_filter': 0, 'public_updates_channel_id': None, 'max_members': 250000, 'features': [], 'afk_timeout': 300, 'splash': None, 'emojis': [{'roles': [], 'require_colons': True, 'name': 'goodnoodle', 'managed': False, 'id': '833659541735604274', 'available': True, 'animated': False}], 'threads': [], 'rules_channel_id': None, 'large': False, 'id': '742107024338452600'}}
DEBUG   ; MainThread           ; 2021-04-21 17:11:54,335; discord.client   358; Dispatching event socket_response
DEBUG   ; MainThread           ; 2021-04-21 17:11:54,335; discord.client   358; Dispatching event guild_available
DEBUG   ; MainThread           ; 2021-04-21 17:11:56,351; discord.client   358; Dispatching event ready
DEBUG   ; MainThread           ; 2021-04-21 17:11:56,351; main             413; in iteration, ID=Gigi's Server (RP)
DEBUG   ; MainThread           ; 2021-04-21 17:11:56,351; Gigi's Server (RP) 1057; logger started
DEBUG   ; MainThread           ; 2021-04-21 17:11:56,351; Gigi's Server (RP) 1072; RoRclient Gigi's Server (RP) initialized
DEBUG   ; RoR_thread_Gigi's Server (RP); 2021-04-21 17:11:56,351; Gigi's Server (RP) 1133; Connecting to server
DEBUG   ; RoR_thread_Gigi's Server (RP); 2021-04-21 17:11:56,351; Gigi's Server (RP)  618; Creating socket...
INFO    ; RoR_thread_Gigi's Server (RP); 2021-04-21 17:11:56,351; Gigi's Server (RP)  627; Created socket...
DEBUG   ; RoR_thread_Gigi's Server (RP); 2021-04-21 17:11:56,351; Gigi's Server (RP)  629; Trying to connect to server 45.156.85.84:50259
INFO    ; RoR_thread_Gigi's Server (RP); 2021-04-21 17:11:56,351; Gigi's Server (RP)  637; Connected to server 45.156.85.84:50259
DEBUG   ; RoR_thread_Gigi's Server (RP); 2021-04-21 17:11:56,366; Gigi's Server (RP)  644; Successfully connected! Sending hello message.
DEBUG   ; RoR_thread_Gigi's Server (RP); 2021-04-21 17:11:56,366; Gigi's Server (RP)  943; S>| HELLO              000:00 (11)
DEBUG   ; Thread-3             ; 2021-04-21 17:11:56,366; Gigi's Server (RP) 1031; R<| HELLO              000:00 (4373)
DEBUG   ; RoR_thread_Gigi's Server (RP); 2021-04-21 17:11:56,366; Gigi's Server (RP)  658; Received server info.
DEBUG   ; RoR_thread_Gigi's Server (RP); 2021-04-21 17:11:56,366; Gigi's Server (RP)  661; Sending our user info.
DEBUG   ; RoR_thread_Gigi's Server (RP); 2021-04-21 17:11:56,366; Gigi's Server (RP)  943; S>| USER_INFO          000:00 (359)
DEBUG   ; Thread-3             ; 2021-04-21 17:11:56,819; Gigi's Server (RP) 1031; R<| WELCOME            024:00 (359)
INFO    ; RoR_thread_Gigi's Server (RP); 2021-04-21 17:11:56,819; Gigi's Server (RP)  689; joined as 'Beholder' on slot -1 with UID 24 and auth 9
DEBUG   ; Thread-3             ; 2021-04-21 17:11:56,819; Gigi's Server (RP) 1031; R<| USER_JOIN          024:00 (359)
DEBUG   ; RoR_thread_Gigi's Server (RP); 2021-04-21 17:11:56,819; Gigi's Server (RP)  943; S>| STREAM_REGISTER    024:10 (272)
stream default: 10
DEBUG   ; RoR_thread_Gigi's Server (RP); 2021-04-21 17:11:56,835; Gigi's Server (RP)  943; S>| STREAM_REGISTER    024:11 (272)
stream chat: 11
Exception in thread RoR_thread_Gigi's Server (RP):
Traceback (most recent call last):
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python39\lib\threading.py", line 954, in _bootstrap_inner
    self.run()
  File "C:\Users\Administrator\Desktop\beholder 1\RoR_client.py", line 1082, in run
    self.bigLoop()
  File "C:\Users\Administrator\Desktop\beholder 1\RoR_client.py", line 1145, in bigLoop
    self.discord.sayInfo("Connected to server %s" % s(serverinfo.servername))
  File "C:\Users\Administrator\Desktop\beholder 1\RoR_client.py", line 456, in sayInfo
    self.__send(msg, "info")
  File "C:\Users\Administrator\Desktop\beholder 1\RoR_client.py", line 424, in __send
    self.main.messageDiscordclient(self.channelID, "[%s] %s" % (prefix, msg))
  File "C:\Users\Administrator\Desktop\beholder 1\services_start.py", line 397, in messageDiscordclient
    bot.loop.create_task(channel.send(message))
AttributeError: 'NoneType' object has no attribute 'send'
DEBUG   ; Thread-3             ; 2021-04-21 17:12:07,929; Gigi's Server (RP) 1031; R<| USER_JOIN          025:00 (359)
DEBUG   ; Thread-3             ; 2021-04-21 17:12:07,976; Gigi's Server (RP) 1031; R<| STREAM_REGISTER    025:10 (272)
DEBUG   ; Thread-3             ; 2021-04-21 17:12:09,226; Gigi's Server (RP) 1031; R<| STREAM_REGISTER    025:11 (272)
DEBUG   ; Thread-3             ; 2021-04-21 17:12:09,320; Gigi's Server (RP) 1031; R<| USER_INFO          025:00 (359)
DEBUG   ; Thread-1             ; 2021-04-21 17:12:35,195; discord.gateway  148; Keeping shard ID None websocket alive with sequence 2.
DEBUG   ; MainThread           ; 2021-04-21 17:12:35,336; discord.client   358; Dispatching event socket_raw_receive
DEBUG   ; MainThread           ; 2021-04-21 17:12:35,351; discord.gateway  432; For Shard ID None: WebSocket Event: {'t': None, 's': None, 'op': 11, 'd': None}
DEBUG   ; MainThread           ; 2021-04-21 17:12:35,351; discord.client   358; Dispatching event socket_response
DEBUG   ; Thread-3             ; 2021-04-21 17:12:46,961; Gigi's Server (RP) 1031; R<| USER_LEAVE         025:00 (23)

```

**Comment by tritonas00** [comment]
```
Exception in thread RoR_thread_Gigi's Server (RP):
Traceback (most recent call last):
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python39\lib\threading.py", line 954, in _bootstrap_inner
    self.run()
  File "C:\Users\Administrator\Desktop\beholder 1\RoR_client.py", line 1082, in run
    self.bigLoop()
  File "C:\Users\Administrator\Desktop\beholder 1\RoR_client.py", line 1145, in bigLoop
    self.discord.sayInfo("Connected to server %s" % s(serverinfo.servername))
  File "C:\Users\Administrator\Desktop\beholder 1\RoR_client.py", line 456, in sayInfo
    self.__send(msg, "info")
  File "C:\Users\Administrator\Desktop\beholder 1\RoR_client.py", line 424, in __send
    self.main.messageDiscordclient(self.channelID, "[%s] %s" % (prefix, msg))
  File "C:\Users\Administrator\Desktop\beholder 1\services_start.py", line 397, in messageDiscordclient
    bot.loop.create_task(channel.send(message))
AttributeError: 'NoneType' object has no attribute 'send'
```

It crashed, i will search and keep you informed

**Comment by tritonas00** [comment]
what discord.py version did you install?

type `py -3 -m pip show discord.py` in command prompt and paste here the output

**Comment by Goetterescu** [comment]
```
Name: discord.py
Version: 1.7.1
Summary: A Python wrapper for the Discord API
Home-page: https://github.com/Rapptz/discord.py
Author: Rapptz
Author-email: None
License: MIT
Location: c:\users\administrator\appdata\local\programs\python\python39\lib\site-packages
Requires: aiohttp
Required-by:
```

**Comment by Goetterescu** [comment]
May it be because I set the Position to "9"? I used it this way in the previous Version and it worked. Im asking because in the server's .auth file it says: 8: bot, robot (blue name, no special privileges)

**Comment by Goetterescu** [comment]
The number before the token and the name

**Comment by tritonas00** [comment]
`1.7.1` is the latest, and i checked again here and it works fine, at least on linux. Let me try on windows

**Comment by tritonas00** [comment]
Checked on windows, works fine also. Can you post your configuration.xml?

**Comment by Goetterescu** [comment]
[configuration.zip](https://github.com/tritonas00/RoRServerBot/files/6352297/configuration.zip)

**Comment by tritonas00** [comment]
Can't see anything wrong. Well i can't think anything else. @WillyB1337 can you help?

**Comment by WillyB1337** [comment]
> Can't see anything wrong. Well i can't think anything else. @WillyB1337 can you help?

I do recall at one time that on my friend’s VPS, I would run both RoRServer & RoRBot but both weren’t able to communicate between each other but when connecting my local hosted RoRBot to the VPS, it would work. 
They weren’t any obvious errors showing up.

**Comment by Goetterescu** [comment]
So what could be a possible solution? RoRbot worked for me in the last version of RoRnet

**Comment by tritonas00** [comment]
I think the problem is on discord side, i cant see any discord traffic in the log https://github.com/tritonas00/RoRServerBot/issues/6#issuecomment-824141348

you should see things like 
![kkk](https://user-images.githubusercontent.com/2660424/115587998-f4481d80-a2d6-11eb-9a87-70edf3a6f928.png)

Also `AttributeError: 'NoneType' object has no attribute 'send'` is alarming

@WillyB1337 how did you install discord.py?

**Comment by Goetterescu** [comment]
Update: the bot is shutting down after about one Minute being online.

**Comment by tritonas00** [comment]
That was fixed 5 days ago, https://github.com/tritonas00/RoRServerBot/commit/06c4ebaf452290fe725a5a6a22bba608f843a32e, make sure you download the latest master branch zip

i also added the `start_bot.bat` launcher

**Comment by Goetterescu** [comment]
Yes, I downloaded the newest Version. I am using that right now

**Comment by tritonas00** [comment]
First things first. Is the `AttributeError: 'NoneType' object has no attribute 'send'` error solved now?

**Comment by Goetterescu** [comment]
no

**Comment by Goetterescu** [comment]
I guess maybe its a issue on my Vserver's side. I will wait for the next RoRbot version. Maybe it will work then. Cause right now i feel like we are just wasting our time here.

**Comment by tritonas00** [comment]
@Goetterescu can you contact me on discord? i have an idea

**Comment by Goetterescu** [comment]
yes

**Comment by tritonas00** [comment]
Problem was wrong discord channel id
